controllers/upload.js

```diff
@@ -0,0 +1,46 @@
+const Jimp = require("jimp");
+const path = require("path");
+const { updateAvatar } = require("../service/user.service");
+const { error } = require("console");
+const fs = require("fs").promises;
+
+const userUpdateAvatar = async ( req, res, next ) => {
+    const { path: tempPath, filename } = req.file;
+    const avatarsDirectory = path.join(
+        process.cwd(),
+        "public/avatars/" + filename
+    );
+
+    const avatar = await Jimp.read(tempPath);
+    await avatar.resize(250, 250).writeAsync(tempPath);
+
+    try {
+        await fs.rename(tempPath, avatarsDirectory);
+    } catch (err) {
+        await fs.unlink(tempPath);
+        console.log(error);
+    }
+
+    if(req.file) {
+        const result = await updateAvatar(req.user.id, avatarsDirectory);
```

Comment:
Otocz to w `try cach` blok i jeśli poszło ok to res. success to user a jeśli nie to w boku catch informację że update avatara się nie powiódł.